systran_storages/storages/s3.py

```diff
@@ -116,18 +116,30 @@ def listdir(self, remote_path, recursive=False, is_file=False):
         delimiter = '/'
         if recursive:
             delimiter = ''
-        list_objects = self._s3.meta.client.list_objects_v2(Bucket=self._bucket_name,
-                                                            Delimiter=delimiter,
-                                                            Prefix=remote_path)
-        if 'CommonPrefixes' in list_objects:
-            for key in list_objects['CommonPrefixes']:
-                listdir[key['Prefix']] = {'is_dir': True}
-        if 'Contents' in list_objects:
-            for key in list_objects['Contents']:
-                listdir[key['Key']] = {'size': key['Size'],
-                                       'last_modified': datetime_to_timestamp(key['LastModified'])}
-                if key['Key'].endswith('/'):
-                    listdir[key['Key']]['is_dir'] = True
+
+        list_objects_args={
+            'Bucket': self._bucket_name,
+            'Delimiter': delimiter,
+            'Prefix': remote_path
+        }
+
+        while True:
+            list_objects = self._s3.meta.client.list_objects_v2(**list_objects_args)
+            if 'CommonPrefixes' in list_objects:
+                for key in list_objects['CommonPrefixes']:
+                    listdir[key['Prefix']] = {'is_dir': True}
+            if 'Contents' in list_objects:
+                for key in list_objects['Contents']:
+                    listdir[key['Key']] = {'size': key['Size'],
+                                           'last_modified': datetime_to_timestamp(key['LastModified']),
+                                           'etag': key['ETag']}
+                    if key['Key'].endswith('/'):
+                        listdir[key['Key']]['is_dir'] = True
+            try:
+                list_objects_args['ContinuationToken'] = list_objects['NextContinuationToken']
```

Comment:
It would be cleaner if you use: `isTruncated` to check if we have more keys to retrieve instead of `try/catch`.

Reply:
indeed - done